.vib/spring-cloud-dataflow/cypress/cypress/integration/spring-cloud-dataflow_spec.js

```diff
@@ -0,0 +1,108 @@
+  cy.get('.signpost-content-body')
+    .invoke('text')
+    .should('match', /.*[0-9].*/);
+  cy.contains('button', 'More info').click();
+  cy.contains('.modal-content', 'Core')
+    .and('contain', 'Dashboard')
+    .and('contain', 'Shell');
+});
+
+it('allows a stream to be created and deployed', () => {
+  const STREAM = 'mongodb | cassandra';
+  const STREAM_APPLICATION = 'Stream application starters for Kafka/Maven';
+
+  importAnApplication(STREAM_APPLICATION);
+  cy.visit('/dashboard');
+  cy.get('[routerlink="streams/list"').click();
+  cy.contains('.btn-primary', 'Create stream(s)').click();
+  cy.get('.CodeMirror-line').type(STREAM);
+  cy.contains('#v-2', 'mongodb').and('contain', 'cassandra');
+  cy.contains('button', 'Create stream(s)').click();
+  cy.get('.modal-content').should('contain', 'Create Stream');
+  cy.fixture('streams').then((stream) => {
+    cy.get('input[placeholder="Stream Name"]').type(
+      `${stream.newStream.name}-${random}`
+    );
+    cy.contains('.btn-primary', 'Create the stream').click();
+    cy.contains(
+      '.datagrid-inner-wrapper',
+      `${stream.newStream.name}-${random}`
+    );
+  });
+  cy.contains('.toast-container', 'successfully');
+  cy.contains('clr-dg-cell', 'UNDEPLOYED')
+    .siblings('clr-dg-cell', 'test-stream')
+    .first()
+    .click();
+  cy.contains('button#btn-deploy', 'Deploy stream').click();
+  cy.contains('button', 'Deploy stream').click();
+  cy.get('.toast-container').should('contain', 'Deploy success');
+});
+
+it('allows a task to be scheduled and destroyed', () => {
+  const CRON_EXPRESSION = '*/5 * * * *';
+  const TASK_APPLICATION = 'Task application starters for Maven';
+  const TASK_TYPE = 'timestamp';
+
+  importAnApplication(TASK_APPLICATION);
+  createATask(TASK_TYPE);
+  cy.get('[routerlink="tasks-jobs/tasks"]').click();
+  cy.contains('clr-dg-cell', 'UNKNOWN')
+    .siblings('clr-dg-cell', 'test-task')
+    .first()
+    .click();
+  cy.contains('button', 'Schedule').click();
+  cy.fixture('schedules').then((schedule) => {
+    cy.get('input[name="example"]')
+      .first()
+      .type(`${schedule.newSchedule.name}-${random}`);
+  });
+  cy.get('input[name="example"]').last().type(CRON_EXPRESSION);
+  cy.contains('button', 'Create schedule(s)').click();
+  cy.contains('.toast-container', 'Successfully');
+  cy.contains('a', 'Tasks').click();
+  cy.contains('clr-dg-cell', 'UNKNOWN')
+    .siblings('clr-dg-cell', 'test-task-')
+    .first()
+    .click();
+  cy.contains('button', 'Destroy task').click();
+  cy.contains('.btn-danger', 'Destroy the task').click();
+  cy.contains('.toast-container', '1 task definition(s) destroyed.');
+});
+
+it('allows importing a task from a file and destroying it ', () => {
+  cy.visit('/dashboard');
+  cy.get('[routerlink="manage/tools"]').click();
+  cy.contains('a', 'Import tasks from a JSON file').click();
+  cy.get('input[type="file"]').selectFile(
+    'cypress/fixtures/task-to-import.json',
```

Comment:
Sorry to be singing always the same song, but would there be an issue if the corner case of importing the same data twice happens? 

Maybe we need to brainstorm this corner case and its possible solutions, as we will encounter it every time we work with non-randomized data.

Reply:
Yes, the same case is happening here. This is why I mentioned this might be a global problem.
In some apps, we have the option to modify the application name which is good (ex. Mediawiki).
However, other apps don't allow for this.
Would it be OK if were leave this for now and brainstorm a global solution we can apply to these types of assets? Spring Cloud Dataflow is not the only one.

Reply:
I've created a task related to this in our internal EPIC.